libseccomp/src/notify.rs

```diff
@@ -5,107 +5,299 @@
 
 use crate::error::ErrorKind::*;
 use crate::error::{Result, SeccompError};
+use crate::{ScmpArch, ScmpFilterContext};
 use libseccomp_sys::*;
-use std::ptr::NonNull;
+use std::io::Error;
 
-pub struct ScmpNotification {
-    req: NonNull<seccomp_notif>,
-    resp: NonNull<seccomp_notif_resp>,
+#[doc(hidden)]
+fn get_errno() -> i32 {
+    if let Some(errno) = Error::last_os_error().raw_os_error() {
+        errno
+    } else {
+        0
+    }
 }
```

Comment:
```suggestion
fn get_errno() -> i32 {
    Error::last_os_error().raw_os_error().unwrap_or(0)
}
```
- private functions are not documented for library targets unless you pass `--document-private-items`

Reply:
Thank you. I fixed it.